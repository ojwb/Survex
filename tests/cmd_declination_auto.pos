( Easting, Northing, Altitude )
(410600.00, 5282000.00,  1234.00 ) 1
(410600.00, 5282000.00,  1234.00 ) a.1
(410593.73, 5282007.72,  1232.95 ) a.2
(410593.74, 5282008.94,  1233.08 ) a.3
(410593.73, 5282007.72,  1232.95 ) b.1
(410593.73, 5282007.72,  1231.95 ) b.2
(410593.73, 5282007.72,  1231.95 ) c.1
(410593.73, 5282007.72,  1230.95 ) c.2
(410594.40, 5282007.26,  1229.13 ) c.3
(410593.73, 5282007.72,  1230.95 ) d.1
(410598.24, 5282107.61,  1230.95 ) d.2
(410595.52, 5282057.68,  1230.95 ) d.2a
(410603.72, 5282007.36,  1230.95 ) d.2b
