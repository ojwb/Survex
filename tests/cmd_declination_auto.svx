; pos=yes warn=2

*cs utm33
*cs out utm33
*fix 1 0410600 5282000 1234
*declination auto 0410600 5282000 1234

*equate 1 a.1
*begin a
*export 1 2
1 2 10.00 320 -06
2 3  1.23 000 +06
*end a

*equate a.2 b.1
*begin b
*export 1 2
; No declination warning here
1 2 1.0 - down
*end b

*equate b.2 c.1
*begin c
*export 1 2
1 2 1.0 - down
; Declination warning here:
2 3 2.0 123 -66
*end c

*equate c.2 d.1
*begin d
*export 1 2
*date 2000.12.25
1 2 100.0 000 0
; Regression test for bug fixed in 1.2.27 - cavern caches the declination,
; but fails to invalidate the cache when the declination coordinates are
; changed, only when the date changed.
*date 2000.12.26
*declination auto 0490600 5282000 1234
1 2a 50.0 000 0
*date 2000.12.27
1 2b 10.0 090 0
; We shouldn't report for this, but did in 1.4.6 and earlier (and convergence
; was reported as "inf").
*declination auto 0490600 5282001 1234
*end d
