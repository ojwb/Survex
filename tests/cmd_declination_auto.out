./cmd_declination_auto.svx:11: warning: No survey date specified - using 0 for magnetic declination
 1 2 10.00 320 -06
./cmd_declination_auto.svx:27: warning: No survey date specified - using 0 for magnetic declination
 2 3 2.0 123 -66
./cmd_declination_auto.svx:6: info: Declination: 1.7dg @ 2000-12-25, grid convergence: -0.9dg
 *declination auto 0410600 5282000 1234
./cmd_declination_auto.svx:39: info: Declination: 2.0dg @ 2000-12-26 / 2.0dg @ 2000-12-27, grid convergence: -0.1dg
 *declination auto 0490600 5282000 1234

Removing trailing traverses...

Concatenating traverses...

Simplifying network...

Calculating network...

Calculating traverses...

Calculating trailing traverses...

Calculating statistics...

Survey contains 13 survey stations, joined by 12 legs.
There are 0 loops.
Total length of survey legs =  175.23m ( 175.23m adjusted)
Total plan length of survey legs =  171.98m
Total vertical length of survey legs =    5.00m
Vertical range = 4.87m (from a.1 at 1234.00m to c.3 at 1229.13m)
North-South range = 107.61m (from d.2 at 5282107.61m to a.1 at 5282000.00m)
East-West range = 9.99m (from d.2b at 410603.72m to d.1 at 410593.73m)
   6 1-nodes.
   4 2-nodes.
   2 3-nodes.
   1 4-node.
There were 2 warning(s).
